_studio/mfx_lib/decode/vc1/src/mfx_vc1_decode.cpp

```diff
@@ -1923,6 +1922,9 @@ mfxStatus   MFXVideoDECODEVC1::FillOutputSurface(mfxFrameSurface1 *surface)
                 surface->Info.PicStruct |= MFX_PICSTRUCT_FIELD_REPEATED;
         }
     }
+
+    SetFrameType(m_pVC1VideoDecoder->m_pStore->GetLastDS()->m_pContext->m_picLayerHeader->PTYPE, *surface);
```

Comment:
Here is a chain of 5 pointers dereferencing... Is there a chance for some of them to be null?

Reply:
@mgonchar , same construction using above in this function (see line 1891).
But I agree, make sense add check.
Right now there pointers can't be null, but if call function from another place possible an issues

Reply:
Considering this function - looks like no. Still it doesn't look like good coding practice.